ddtrace/llmobs/_integrations/google_genai.py

```diff
@@ -16,3 +49,147 @@ def _set_base_span_tags(
+            code = _get_attr(executable_code, "code", "")
+            message["content"] = "[executable code ({language}): {code}]".format(language=language, code=code)
+            return message
+
+        code_execution_result = _get_attr(part, "code_execution_result", None)
+        if code_execution_result:
+            outcome = _get_attr(code_execution_result, "outcome", "OUTCOME_UNSPECIFIED")
+            output = _get_attr(code_execution_result, "output", "")
+            message["content"] = "[code execution result ({outcome}): {output}]".format(outcome=outcome, output=output)
+            return message
+
+        thought = _get_attr(part, "thought", None)
+        # thought is just a boolean indicating if the part was a thought
+        if thought:
+            message["content"] = "[thought: {}]".format(thought)
+            return message
+
+        return {"content": "Unsupported file type: {}".format(type(part)), "role": role}
+
+    def _extract_input_message(self, args, kwargs, config):
+        messages = []
+        # system instruction in config
+        if config is not None:
+            system_instruction = _get_attr(config, "system_instruction", None)
+            if system_instruction is not None:
+                normalized_sys = normalize_contents(system_instruction)
+
+                for content in normalized_sys:
+                    role = content.get("role") or "system"
+                    parts = content.get("parts", [])
+
+                    for part in parts:
+                        message = self._extract_message_from_part_google_genai(part, role)
+                        messages.append(message)
+        # user input messages
+        contents = get_argument_value(args, kwargs, -1, "contents")
+        normalized_contents = normalize_contents(contents)
+        for content in normalized_contents:
+            role = content.get("role") or "user"
+            parts = content.get("parts", [])
+
+            for part in parts:
+                message = self._extract_message_from_part_google_genai(part, role)
+                messages.append(message)
+        return messages
+
+    def _extract_output_message(self, response):
+        if not response:
+            return [{"content": ""}]
+        # streamed responses will be a list of chunks
+        if isinstance(response, list):
+            message_content = []
+            tool_calls = []
+            role = "model"
+            # response is a list of GenerateContentResponse chunks
+            for r in response:
+                messages = self._process_response(r)
+                for message in messages:
+                    message_content.append(message.get("content", ""))
+                    tool_calls.extend(message.get("tool_calls", []))
+            message = {"content": "".join(message_content), "role": role}
+            if tool_calls:
+                message["tool_calls"] = tool_calls
```

Comment:
I'd like this method to be on the contrib/integration utils side, i.e. construct the message from the chunks first before passing it onto llmobs_set_tags(). The integration class shouldn't have to worry about parsing streamed chunks etc